src/server/src/modules/facultyMember.ts

```diff
@@ -81,3 +83,63 @@ export const getFacultyMemberProfile = (id: number) => {
         .leftJoinAndSelect('department.college', 'college')
         .getOne();
 };
+
+/**
+ * @description Get a list of jobs posted by a faculty member
+ * @param {number} facultyMemberId - Id of faculty member
+ * @returns Promise
+ */
+export const getPostedJobs = async (facultyMemberId: number) => {
+    // Check if a faculty member with the given id exists.
+    const facultyMember = await FacultyMember.findOne(facultyMemberId);
+    if (!facultyMember) return undefined;
+
+    // Returns all jobs he posted.
+    return getRepository(Job)
+        .createQueryBuilder('job')
+        .where({ facultyMemberId: facultyMemberId })
+        .leftJoinAndSelect('job.department', 'department')
+        .leftJoinAndSelect('department.college', 'college')
+        .getMany();
+};
+
+
+
+/**
+ * @description Get a list of students who applied to a job.
+ * @param {number} facultyMemberId - Id of faculty member
+ * @param {number} jobId - Id of the job
+ * @returns Promise
+ */
+export const getApplicants = async (facultyMemberId: number, jobId: number) => {
```

Comment:
All good but take a look at the applyToJob function from the students module. There you will see how to instead of throwing an error, return an object that stores an error message for the caller of the API.

Reply:
Done.